python/py-pyobjc/Portfile

```diff
@@ -28,7 +30,7 @@ long_description    The PyObjC project aims to provide a bridge between \
                     Python based functionality.
 homepage            https://pyobjc.readthedocs.io
 
-python.versions     37 38 39 310 311 312
+python.versions     38 39 310 311 312
```

Comment:
You cannot just remove the py37 subport as there are still other ports that depend on it:
```
❯ port echo depends:py37-pyobjc
py37-imagecodecs
py37-zope-hookable
py37-tensorflow-addons
```

Reply:
Good point! I will make a separate PR on removing the py37 subports first then. Converting this one to draft in the meanwhile

Reply:
that change should be made in this PR please. Also, please note that this has been done recursively - so you'll need to check if any of the three ports listed above still have dependencies, and, if so, continue that process until there isn't anything showing up anymore.

Reply:
Got it :) What do I do about the PR title / commit message title – should I mention those other ports as well or is it sufficient to mention them in the body of the commit message?

Reply:
CI failed after I touched `py-tensorflow-addons` because its py38 subport (which I did not remove at first) depended on now-removed `py38-tensorflow`, so I removed that subport and its (only) dependent, `py38-tensor2tensor`.

Reply:
these should be in different commits but in the same PR. 
Also, why did you remove PY38 subports; that Python version is still supported so they should remain.

Reply:
Okay, I will redo the commits.

I can put back the py38 subports, but they will still not work since `py38-tensorflow` does not exist anymore ([dropped in May](https://github.com/macports/macports-ports/commit/fb29cdd5712878122fe6284407d0d977a1194154)).

Reply:
oh, I see - that was already broken indeed. Yes, sure just remove the py38 subports then; please refer to the commit that incorrectly dropped `py38-tensorflow` in that commit message.

Reply:
I just double-checked the dependency on `py38-tensorflow` in the `py-tensorflow-addons` Portfile before removing `38` and I am unsure this is a hard dependency after all. There is this line (a build dependency):

`path:${python.pkgd}/tensorflow:py${python.version}-tensorflow`

If I have understood correctly, the `path:` thing means that one could meet the dependency by installing `tensorflow` from somewhere else, for example an old version from PIP. There is also a port `py-tensorflow1` that does have a 38 subport. It is incompatible with my system, but I guess there is a good chance it will install to `${python.pkgd}`, because it conflicts with the one without the `1` suffix. I am trying to push a commit that replaces this dependency line with the following:

```
if {${python.version} eq 38} {
    depends_build-append \
        path:${python.pkgd}/tensorflow:py${python.version}-tensorflow1
} else {
    depends_build-append \
        path:${python.pkgd}/tensorflow:py${python.version}-tensorflow
}
```
If that passes CI I guess I can keep the py38 subports?

Reply:
just leave the py38 subport; it's not something you changed in this PR so it's up to the maintainer to fix a possible problem. The fact that the CI will not pass because of that is fine. Please change as little as possible that is needed for main change in the PR to work (i.e., the update to the latest upstream version requires to drop the PY37 subport and thus dependents as well).